### DISCUSSION TITLE
Go Back and Go Forward

### DISCUSSION THREAD

**Discussion by SethFalco** [Kong/insomnia#4810] [Ideas]
When a user performs back or forward, the user should go up/down a stack of requests that access recently.

This refers to the "Go Back" and "Go Forward" implemented in many applications, namely browsers. This is also very common in Electron applications in general such as Visual Studio Code, Jellyfin Media Player, Element, Tutanota, FreeTube, etc. (Other than the browser, none of these have dedicated buttons for back/forward in the UI either, but the mouse buttons and keyboard shortcut still works.

The interface doesn't currently display buttons to perform this function, which is a separate discussion.

But for example, many mice including mine ([Logitech MX Master 3](https://www.logitech.com/en-us/products/mice/mx-master-3.html)) have a physical back and forward button, so these actions are still possible.

Many users are also familiar with the shortcuts `ALT` + `←` and `ALT` + `→` which are associated with "Go Back" and "Go Forward" respectively. ([Source](https://en.wikipedia.org/wiki/Table_of_keyboard_shortcuts#Browsers_/_Go_menu
))

It'd be great to see this implemented in Insomnia for ease of use when jumping between multiple queries. I do this regularly when I need to copy and paste parts of the output of one request for the input of another request for example.

I'm happy to try to implement this myself if you're into it but aren't willing to take it up yourselves, tips welcome for getting started.

**Comment by endo64** [comment]
This would be a great feature!

**Comment by LNTR** [comment]
Would be nice if we can get the alt + arrow shortcut keys for navigation
